### ISSUE TITLE
Add flag to opt out one test from parallel testing

### ISSUE THREAD

**Issue by dns2utf8** [rust-lang/cargo#2662]
Hi all

I am not certain this is the right place, because the parsing of the test files appears to be done in [rustc](http://doc.rust-lang.org/1.1.0/src/test/lib.rs.html#336) as well.

I had a problem while testing the threadpool-crate because all test run in parallel and all of them use the whole cpu.
Would it be possible to add a additional flag for test like this? Maybe something like this:
```rust
#[test]
#[no_parallel]
fn test_heavy_load_on_cpu_or_gpu_or_global_state() {
  // working hard here ...
}
```

The implementation would have to queue these test and run them sequencally.

The flag `no_parallel` seems very expressiv, but I prefere a positive keyword like `single` or `individual` or something like this.

What do you think?
Regards

**Comment by alexcrichton** [comment]
Thanks for the report! This is actually more of a rustc bug than a Cargo bug (as the compiler is what processes these annotations), so you may want to open a bug over at rust-lang/rust.

I know I've wanted the ability to specify something like this before, though! One trick is to set `RUST_TEST_THREADS=1` in the environment and that'll make sure the tests are single-threaded, but ideally there'd be a way to encode that in the source itself.

**Comment by dns2utf8** [comment]
Okay, i will open another issue in Rust. In my case running all tests in a chain kills the performance oft the tests. So flagging just 2 out of 20+ tests would be very handy.

**Comment by dns2utf8** [comment]
Added Issue to Rust: https://github.com/rust-lang/rust/issues/33519

**Comment by alexcrichton** [comment]
Thanks!
